tools/forms/upload_raw.py

```diff
@@ -0,0 +1,47 @@
+from typing import Set
+
+from django import forms
+from django.utils.safestring import mark_safe
+from django.utils.translation import gettext as _
+
+
+RAW_ERRORS = {
+    "no_org": _("Organization code(s) for raw does not exist in our database"),
+    "decoding": _("File could not be decoded"),
+}
+
+
+class UploadRawForm(forms.Form):
+    mime_types = forms.CharField(
+        label=_("Mime types"),
+        help_text=mark_safe(
+            _(
+                '<p>Add a set of mime types, separated by commas, for example:</p><p><i>"text/html, image/jpeg"</i> or '
+                '<i>"boefje/dns-records"</i>.</p><p>Mime types are used to match the correct normalizer to a raw file. '
+                'When the mime type "boefje/dns-records" is added, the normalizer expects the raw file to contain dns '
+                "scan information.</p>"
+            )
+        ),
+        required=True,
+        widget=forms.TextInput(attrs={"placeholder": "text/html, image/jpeg, ..."}),
+    )
+    raw_file = forms.FileField(
+        label=_("Upload raw file"),
+        allow_empty_file=False,
+    )
+
+    def clean_raw_file(self):
+        raw_file = self.cleaned_data["raw_file"]
+
+        try:
+            raw_file.read().decode("UTF-8")
```

Comment:
Do we assume that raw files are always (text) files encoded with UTF8 encoding? This way you can't upload any file format as suggested by the MIME type field

Reply:
Ah fair point

Reply:
Fixed